components/scream/cime_config/buildnml

```diff
@@ -199,281 +194,264 @@ def evaluate_selectors(element, case, ez_selectors):
-                                       'interleave': ['string', None],
-                                       'prec_tester': ['integer', None],
-                                       'sel_regex_no_match': ['integer', None],
-                                       'testval': ['integer', None],
-                                       'testval2': ['integer', None]}}}
-    >>> ######################################################################
-    >>> test_metadata_override = '''
-    ... <namelist_defaults>
-    ...   <selectors>
-    ...     <selector name="hgrid" case_env="ATM_GRID"/>
-    ...   </selectors>
-    ...   <file name="test.yaml" format="yaml">
-    ...     <testval valid_values="1,2,3">1</testval>
-    ...     <testval hgrid="ne4ne4" valid_values="1,2,3,4">2</testval>
-    ...     <testval hgrid="no_match" valid_values="1,2,3,4,5">3</testval>
-    ...     <testval2 valid_values="1,2,3">1</testval2>
-    ...     <testval2 hgrid="ne4ne4">2</testval2>
-    ...     <testval2 hgrid="no_match">3</testval2>
-    ...   </file>
-    ... </namelist_defaults>
-    ... '''
-    >>> _create_raw_xml_file_impl(case, test_metadata_override)[1]
-    {'test.yaml': {'testval': ['integer', '1,2,3,4'], 'testval2': ['integer', '1,2,3']}}
-    """
-    root = ET.fromstring(defaults_xml)
-
-    selectors = None
-    selectors_elem = None
-    metadata = {}
-    for child in root:
-        if selectors is None:
-            expect(child.tag == "selectors", "Unexpected child.tag {}".format(child.tag))
-            selectors_elem = child
-
-            selectors = {}
-            for selector in child:
-                expect(selector.tag == "selector", "Expected selector tag, not {}".format(selector.tag))
-
-                selector_name  = selector.attrib["name"]
-                expect(selector_name not in METADATA_ATTRIBS,
-                       "Cannot use '{}' as a selector name, that name is reserved for defining metdata".format(selector_name))
-                selector_env   = selector.attrib["case_env"]
-                if "regex" in selector.attrib:
-                    selector_regex = selector.attrib["regex"]
-                else:
-                    selector_regex = "(.*)" # Just grab the whole thing
-
-                selectors[selector_name] = (selector_env, selector_regex)
-
-        else:
-            expect(child.tag == "file", "Unexpected child.tag {}".format(child.tag))
-            expect("name" in child.attrib, "file element missing required 'name' attribute")
-            expect("format" in child.attrib, "file element missing required 'format' attribute")
-
-            metadata[child.attrib["name"]] = evaluate_selectors(child, case, selectors)
-
-    if selectors_elem is not None:
-        root.remove(selectors_elem)
-
-    return ET.tostring(root, encoding="unicode"), metadata
+    #  """
+    #  >>> case = MockCase({'ATM_GRID':'ne4ne4', 'SCREAM_CMAKE_OPTIONS':'FOO=ON SCREAM_NUM_VERTICAL_LEV 128 BAR=OFF'})
```

Comment:
All these good tests are commented out. I assume you didn't want to worry about them as you refactored things, but they should be re-enabled and fixed-up now that you are done.

Also, the new "inherit" and "locked" mechanisms will need tests.